samza-core/src/main/scala/org/apache/samza/metrics/reporter/MetricsSnapshotReporter.scala

```diff
@@ -83,15 +82,18 @@ class MetricsSnapshotReporter(
   }
 
   def stop = {
-    info("Stopping producer.")
 
-    producer.stop
+    // Scheduling an event with 0 delay to ensure flushing of metrics one last time before shutdown
+    executor.schedule(this,0, TimeUnit.SECONDS)
```

Comment:
is this a "best-effort" flush? if not, we should probably schedule this from the current thread synchronously. I believe a best effort flush might be good enough for most cases since the timeout is 60s